lib/collection/src/collection.rs

```diff
@@ -308,6 +325,16 @@ impl Collection {
 
 impl Drop for Collection {
     fn drop(&mut self) {
-        self.stop().unwrap(); // Finishes update tasks right before destructor stucks to do so with runtime
+        self.stop().unwrap(); // Finishes update tasks right before destructor stuck to do so with runtime
+
+        // The drop could be called from the tokio context, e.g. from perform_collection_operation method.
+        // Calling remove from there would lead to the following error in a new version of tokio:
+        // "Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context."
+        // So the workaround for move out the runtime handler and drop it in the separate thread.
+        // The proper solution is to reconsider the collection to be an owner of the runtime
+        let handle = self.runtime_handle.take();
+        thread::spawn(move || {
+            drop(handle);
+        });
```

Comment:
Since the `runtime_handle` is shared between collections, why should we drop it on collection destruction?

Reply:
It is not shared. It is actually an optimizer runtime handler which is collection specific

Reply:
oh, yes, you are right.